### PR TITLE
Improve the error message for missing required inputs

### PR DESCRIPTION
**Pull request by ninghu**
# Description

This PR includes the following changes:

- Refactored the code for column validation by separating target validation and evaluator validation.
- Fixed a regression where an error was not raised when the target doesn't generate columns needed by the evaluator.
- Instead of raising missing inputs error for a single evaluator,  all evaluators are now reported at once, enhancing the customer experience. Below is one example:

image

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by diondrapeck — APPROVED** [review]

**Comment by diondrapeck** [comment]
This looks great - thanks!

File: sdk/evaluation/azure-ai-evaluation/tests/unittests/test_evaluate.py [hunk 1]
```diff
@@ -143,20 +143,25 @@ def test_evaluate_missing_required_inputs(self, missing_columns_jsonl_file):
...
+    def test_target_not_generate_required_columns(self, questions_file):
         with pytest.raises(EvaluationException) as exc_info:
-            # target_fn will generate the "response", but not ground truth.
+            # target_fn will generate the "response", but not "ground_truth".
             evaluate(data=questions_file, evaluators={"g": F1ScoreEvaluator()}, target=_target_fn)
 
-        assert "Missing required inputs for evaluator g : ['ground_truth']." in exc_info.value.args[0]
+        expected_message = "Some evaluators are missing required inputs:\n" "- g: ['ground_truth']\n"
+
+        expected_message2 = "Verify that the target function is generating the necessary columns for the evaluators."
```
Comment (singankit) [inline sdk/evaluation/azure-ai-evaluation/tests/unittests/test_evaluate.py:161]: Would it be possible to print which columns were to be generated by target but target did not generate them ?

Comment (ninghu) [reply sdk/evaluation/azure-ai-evaluation/tests/unittests/test_evaluate.py:161]: That's a good idea. However, it's difficult to deduce the correct target columns, as the output from the target can be very dynamic. What we can do is print the columns generated by the target, so customers can quickly compare them with the evaluator's missing fields. That can help customers to identify the missing inputs root cause more quickly.

Comment (ninghu) [reply sdk/evaluation/azure-ai-evaluation/tests/unittests/test_evaluate.py:161]: This looks better?
image

**Pull request merged**